care/facility/tests/test_facilityuser_api.py

```diff
@@ -25,3 +25,28 @@ def test_get_queryset_with_prefetching(self):
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertNumQueries(2)
+
+    def test_link_new_facility(self):
+        response = self.new_request(
+            (f"/api/v1/getallfacilities",),
+            {"get": "list"},
+            FacilityViewSet,
+            self.users[0],
+            {"facility_external_id": self.facility.external_id},
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertNumQueries(2)
+
+    def test_link_existing_facility(self):
+        print(self.users[0])
+        response = self.new_request(
+            (f"/api/v1/getallfacilities",),
+            {"get": "list", "exclude_user": self.users[0].username},
+            FacilityViewSet,
+            self.users[0],
+            {"facility_external_id": self.facility.external_id},
+        )
```

Comment:
use self.client.get instead of self.new_request

Reply:
Updated